test/unit/tracer-test.js

```diff
@@ -18,60 +18,98 @@
+    carrier[CARRIER_TRANSACTION] = 'foo';
+    carrier[CARRIER_LEVEL] = 'All';
 
     const spanContext = new APMTracer().extract(Constants.FORMAT_HTTP_HEADERS, carrier);
     t.deepEquals(spanContext.getTraceId(), '1');
-    t.deepEquals(spanContext.getId(), '2');
+    t.deepEquals(spanContext.getConsumerCorrelationId(), '2');
+    t.deepEquals(spanContext.getTransaction(), 'foo');
+    t.deepEquals(spanContext.getLevel(), 'All');
     t.end();
 });
 
 test('test extract FORMAT_TEXT_MAP', (t) => {
     const carrier = {};
     carrier[CARRIER_TRACE_ID] = '1';
-    carrier[CARRIER_SPAN_ID] = '2';
+    carrier[CARRIER_CORRELATION_ID] = '2';
+    carrier[CARRIER_TRANSACTION] = 'foo';
+    carrier[CARRIER_LEVEL] = 'All';
 
     const spanContext = new APMTracer().extract(Constants.FORMAT_TEXT_MAP, carrier);
     t.deepEquals(spanContext.getTraceId(), '1');
-    t.deepEquals(spanContext.getId(), '2');
+    t.deepEquals(spanContext.getConsumerCorrelationId(), '2');
+    t.deepEquals(spanContext.getTransaction(), 'foo');
+    t.deepEquals(spanContext.getLevel(), 'All');
     t.end();
 });
 
 test('test extract unknown format', (t) => {
     const carrier = {};
     carrier[CARRIER_TRACE_ID] = '1';
-    carrier[CARRIER_SPAN_ID] = '2';
+    carrier[CARRIER_CORRELATION_ID] = '2';
+    carrier[CARRIER_TRANSACTION] = 'foo';
+    carrier[CARRIER_LEVEL] = 'All';
 
     const spanContext = new APMTracer().extract('unknown', carrier);
     t.deepEquals(spanContext.getTraceId(), undefined);
-    t.deepEquals(spanContext.getId(), undefined);
+    t.deepEquals(spanContext.getConsumerCorrelationId(), undefined);
+    t.deepEquals(spanContext.getTransaction(), undefined);
+    t.deepEquals(spanContext.getLevel(), undefined);
     t.end();
 });
 
 test('test inject FORMAT_HTTP_HEADERS', (t) => {
-    const spanContext = new APMSpanContext(2, 1);
+    const spanContext = new APMSpanContext(2, 1, null, null, 'foo', 'All');
     const carrier = {};
 
     new APMTracer().inject(spanContext, Constants.FORMAT_HTTP_HEADERS, carrier);
-    t.deepEquals(carrier[CARRIER_SPAN_ID], spanContext.getId());
+    t.ok(carrier[CARRIER_CORRELATION_ID]);
     t.deepEquals(carrier[CARRIER_TRACE_ID], spanContext.getTraceId());
+    t.deepEquals(carrier[CARRIER_TRANSACTION], spanContext.getTransaction());
+    t.deepEquals(carrier[CARRIER_LEVEL], spanContext.getLevel());
+    t.end();
+});
+
+test('test extract inject', (t) => {
+    const tracer = new APMTracer();
+    const extractCarrier = {};
+    extractCarrier[CARRIER_CORRELATION_ID] = '12345';
+    extractCarrier[CARRIER_TRACE_ID] = '555';
+    extractCarrier[CARRIER_LEVEL] = 'All';
+    extractCarrier[CARRIER_TRANSACTION] = 'foo';
+
+    const extractSpanContext = tracer.extract(Constants.FORMAT_HTTP_HEADERS, extractCarrier);
+
+    const rootSpan = tracer.startSpan('root', { childOf: extractSpanContext });
+    let injectCarrier = {};
+    tracer.inject(rootSpan.context(), Constants.FORMAT_HTTP_HEADERS, injectCarrier);
```

Comment:
Could you add a child span in here, and do the inject from that, just to ensure state is being passed between the top level root span and the child, before being injected into the outbound message?